src/monattack.cpp

```diff
@@ -916,6 +949,9 @@ void mattack::fungus(monster *z, int index)
 
 void mattack::fungus_haze(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
I left this unchanged for Mycus folks as they wouldn't significantly change their behavior.  If you're Mycus and therefore friendly, you ought to be immune to the haze, so just walk through.  If you're not yet immune, having exposure grant the immunity seems reasonable and I can PR that, no problem.  Either way, the flowers' job is to haze the area so they'll keep pumping out the haze.